File: load_balancer.js
const express = require('express')
const app = express()
const servers = [
    'http://localhost:3000',
    'http://localhost:3001',
    'http://localhost:3002',
    'http://localhost:3003',
    'http://localhost:3004',
    'http://localhost:3005',
    'http://localhost:3006',
    'http://localhost:3007',
    'http://localhost:3008',
    'http://localhost:3009',
]

let current = 0;


let request = 0;
const handler = async(req,res)=>{
    const {method, url, headers, body} = req

    const server = servers[current]

    current === servers.length-1? current = 0 : current++

    try{
        const response = await fetch(server,{
            method:method,
            body:body,
            headers:headers
        })
        let resp = await response.text()
        request++
        res.send(resp)
        console.log(`#${request} ${resp}`)
    }catch(err){
        res.status(500).send('Server Error!')
    }
}

app.use((req,res)=>{handler(req,res)})

app.listen(8080)

File: load_balancer.go
package main

import (
	"fmt"
	"io"
	"log"
	"net/http"
)

func main(){
	fmt.Printf("Starting server on 8080\n")

	servers := [10]string{"http://localhost:3000","http://localhost:3001","http://localhost:3002","http://localhost:3003","http://localhost:3004","http://localhost:3005","http://localhost:3006","http://localhost:3007","http://localhost:3008","http://localhost:3009"}
	
	serverNo := 0
	http.HandleFunc("/",func(w http.ResponseWriter, r *http.Request){
		fmt.Println("%s",servers[serverNo])
		res, err := http.Get(servers[serverNo])

		serverNo = serverNo + 1

		if(serverNo == 10){
			serverNo = 0
		}

		if err != nil{
		}

		resBody, err := io.ReadAll(res.Body)
		if err != nil{
		}
		fmt.Fprintf(w,"%s",resBody)
	})
	if err:= http.ListenAndServe(":8080",nil); err != nil {
		log.Fatal(err)
	}
}